DOC Ensures that sklearn.utils.validation.indexable passes numpydoc validation

majauhar opened the pull request:
Co-authored-by: miwojc <miwojc@protonmail.com>

<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/main/CONTRIBUTING.md
-->

#### Reference Issues/PRs
Addresses #21350
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->


#### What does this implement/fix? Explain your changes.


#### Any other comments?
This is my and @miwojc second pull request made during #DataUmbrella sprint.

<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

majauhar committed:
Add numpydoc validation for sklearn.utils.validation.indexable

Co-authored-by: miwojc <miwojc@protonmail.com>

glemaitre reviewed (commented):
Thanks for the PR. I propose a small suggestion to follow what is proposed in `numpydoc`:

- since we return `result`, we can use it as a name of a variable
- we can specify what to expect as type in the returned list.
- adding the specificity of passing `None` through in the results.

majauhar committed:
Update sklearn/utils/validation.py

Modifies the Returns section in the docstring of indexable() function to make it more informative and clear.

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
